crowbar_framework/app/controllers/installer/upgrades_controller.rb

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Installer
+  class UpgradesController < ApplicationController
+    def prepare
+      if request.post?
+        url = begin
+          service_object = CrowbarService.new(Rails.logger)
+
+          # transition nodes to "crowbar_upgrade"
```

Comment:
IMHO this comment is pretty useless as it's already obious with the method name.